src/main/java/core/basesyntax/Calculator.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Calculator {
+
+    public void calculateStatistics(Report report, ReportItem reportItem) {
+        if (Name.BUY.equals(reportItem.getName())) {
+            report.addBuy(reportItem.getValue());
+        } else {
+            report.addSupply(reportItem.getValue());
+        }
+    }
```

Comment:
The `calculateStatistics` method seems to be doing more than one thing: it is both checking the type of the `ReportItem` and updating the `Report`. Consider dividing this logic into separate methods to adhere to the single responsibility principle.